src/EditorFeatures/Core/Shared/Utilities/IWorkspaceContextService.cs

```diff
@@ -32,13 +34,16 @@ internal sealed class DefaultWorkspaceContextService : IWorkspaceContextService
         /// </summary>
         public const string LspEditorFeatureFlagName = "Roslyn.LSP.Editor";
 
+        private readonly IGlobalOptionService _globalOptionsService;
+
         [ImportingConstructor]
         [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
-        public DefaultWorkspaceContextService()
+        public DefaultWorkspaceContextService(IGlobalOptionService globalOptionsService)
         {
+            _globalOptionsService = globalOptionsService;
         }
 
-        public bool IsInLspEditorContext() => false;
+        public bool IsInLspEditorContext() => _globalOptionsService.GetOption(LspOptions.LspEditorFeatureFlag);
```

Comment:
This is the service exported for the interactive window, so we need to ensure we check the option here. (The VS version of this service only applies to the host workspace).